File: src/sources/php/check_io.php
<?php

$stdOut = fopen('php://stdout', 'w');
$stdErr = fopen('php://stderr', 'w');
$stdIn = fopen('php://stdin', 'r');

$argv = get_defined_vars();
/*
foreach ($argv as $arg) {
    fwrite($stdOut, "arg: " . $arg . PHP_EOL);
}
*/

var_dump(glob('/persist/*'));

foreach ($_ENV as $key => $value) {
    fwrite($stdOut, "env: " . $key . "=" . $value . PHP_EOL);
}

$stdinContent = fread($stdIn, 1024);
if ($stdinContent) {
    fwrite($stdOut, "from stdin: " . $stdinContent . PHP_EOL);
}

fwrite($stdOut, "stdout message!\n");
fwrite($stdErr, "stderr message!\n");

$exitCode = (count($argv) > 1) ? 42 : 0;
exit($exitCode);

File: src/sources/python/check_io.py
import os
import sys

def main():
    for arg in sys.argv:
        sys.stdout.write(f"arg: {arg}\n")

    for key, value in os.environ.items():
        sys.stdout.write(f"env: {key}={value}\n")

    try:
        buf = sys.stdin.read(1024)
        if buf:
            sys.stdout.write("from stdin: " + buf + "\n")
    except Exception as e:
        sys.stderr.write(f"Error reading stdin: {e}\n")

    sys.stdout.write("stdout message!\n")
    sys.stderr.write("stderr message!\n")

    code = 42 if len(sys.argv) > 1 else 0
    sys.exit(code)

if __name__ == "__main__":
    main()
